packages/python-sdk/e2b/sandbox_sync/filesystem/filesystem.py

```diff
@@ -122,27 +124,85 @@ def write(
         :param request_timeout: Timeout for the request
         :return: Information about the written file
         """
-        if isinstance(data, TextIOBase):
-            data = data.read().encode()
+
+    @overload
+    def write(
+        self,
+        files: List[WriteEntry],
+        user: Optional[Username] = "user",
+        request_timeout: Optional[float] = None,
+    ) -> List[EntryInfo]:
+        """
+        Writes multiple files.
+
+        :param files: list of files to write 
+        :param user: Run the operation as this user
+        :param request_timeout: Timeout for the request
+        :return: Information about the written files
+        """
+
+    def write(
+        self,
+        path_or_files: str | List[WriteEntry],
+        data_or_user: WriteData | Username = "user",
+        user_or_request_timeout: Optional[float | Username] = None,
+        request_timeout_or_none: Optional[float] = None
+    ) -> EntryInfo | List[EntryInfo]:
+        """
+        Writes content to a file on the path.
+        When writing to a file that doesn't exist, the file will get created.
+        When writing to a file that already exists, the file will get overwritten.
+        When writing to a file that's in a directory that doesn't exist, you'll get an error.
+        """
+
+        path, write_files, user, request_timeout  = None, [], "user", None
+        if isinstance(path_or_files, str):
+            if isinstance(data_or_user, list):
+                raise Exception("Cannot specify path with array of files")
```

Comment:
Let's change this error string to be more explicit what user is doing wrong here:
```
Cannot specify both path and array of files.
```

maybe we could be even more helpful here:
```
Cannot specify both path and array of files. You have to specify either path and data for a single file or an array for multiple files.
```